tests/core/test_core_memory_resources.cpp

```diff
@@ -78,8 +78,9 @@ class core_memory_resource_test
 
         // Remove a couple of elements from the vectors.
         for (int i : {26, 38, 25}) {
-            std::remove(reference_vector.begin(), reference_vector.end(), i);
-            std::remove(test_vector.begin(), test_vector.end(), i);
+            (void)std::remove(reference_vector.begin(), reference_vector.end(),
+                              i);
+            (void)std::remove(test_vector.begin(), test_vector.end(), i);
```

Comment:
I think this is the only part I don't like, I'd go for `static_cast<void>(...)` in this case.

Reply:
I don't know... I'm not trying to cast the results into anything. I'm just trying to explicitly state that I don't care about the return value. Since MSVC is very vocal about these functions being `[[nodiscard]]`... Which, by the way, is not standard as far as I can see... 😦 

https://en.cppreference.com/w/cpp/algorithm/remove

From all the examples I've seen so far, everybody uses this `(void)` notation in such cases. I haven't seen anybody using `static_cast<void>(...)` for this... 😕

Reply:
It really makes no sense to me why MSVC would mark this `[[nodiscard]]` in their standard library. The return value of this function seems... Not particularly useful.

Anyway, that doesn't matter. In the end, `(void)` and `static_cast<void>` are equivalent, and both will do the job. In my opinion, the `static_cast` version is more C++-like, although I must admit that the C version is more readable.

To be honest I am not sure it matters. Let's merge it in as is.